packages/fuchsia_remote_debug_protocol/lib/src/common/logging.dart

```diff
@@ -0,0 +1,126 @@
+// Copyright 2018 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:core';
+
+/// Determines the level of logging.
+///
+/// Verbosity is increasing from zero (none) to four (fine). The fifth level
+/// (all) logs everything.
+enum LoggingLevel {
+  /// Logs no logs.
+  none,
+
+  /// Logs severe logs at the most.
+  severe,
+
+  /// Logs warning logs at the most.
+  warning,
+
+  /// Logs info logs at the most.
+  info,
+
+  /// Logs fine logs at the most.
+  fine,
+
+  /// Logs everything.
+  all,
+}
+
+/// Synchronously logs a `Log.`
+typedef void LoggingFunction(Log log);
+LoggingFunction _defaultFormatter = (Log log) {
+  print('[${log.levelName}] -- ${log.time}: ${log.message}');
+};
+
+/// Represents a Log created by the logger.
+///
+/// Includes a message, the time the message was created, the level of the log
+/// as an enum, the name of the level as a string, and a tag. This class is used
+/// to print from the global logging function defined in
+/// `Logger.loggingFunction` (a function that can be user-defined).
+class Log {
+  /// The actual log message.
+  final String message;
+
+  /// The time the log message was created.
+  final DateTime time;
+
+  /// The level of this log.
+  final LoggingLevel level;
+
+  /// The human readable level of this log.
+  final String levelName;
+
+  /// The tag associated with the message.
```

Comment:
We probably want to explain that this is the value of `[Logger.tag]` of the `Logger` that emitted this message.

Reply:
Done.